pkg/services/rpcsrv/server.go

```diff
@@ -2441,9 +2447,9 @@ func (s *Server) subscribe(reqParams params.Params, sub *subscriber) (interface{
 	}
 
 	s.subsLock.Lock()
```

Comment:
Move it after the `select`?

Reply:
I think we can't do it, because we need this lock before select due to [this line](https://github.com/nspcc-dev/neo-go/pull/2809#discussion_r1026106753). These two operations (subscription here and subscription in the end of `handleSubEvents`) must be performed sequentially, and this code should have an ability to escape in case if server was shutted down.

Reply:
Or if we'd like to move this lock here and remove `subsLock` [there](https://github.com/nspcc-dev/neo-go/pull/2809#discussion_r1026106753), then shutdown check must be performed a bit later, right after the `subsCounterLock` taken.

Reply:
See the last commit.